lib/factorial.rb

```diff
@@ -1,6 +1,18 @@
 # Computes factorial of the input number and returns it
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) 
+# Space complexity: O(1)
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError if number.nil?
+  if number == 0 || number == 1
+    return 1
+  end
```

Comment:
What would happen if I put `factorial("5")` and would that be a reasonable case to guard against/make work?